docs/guides/building-forms.md

```diff
@@ -82,6 +82,53 @@ With these validation checks in place, when a user tries to submit an empty fiel
 
 ![form validation](/docs/guides/images/form-validation.jpg)
 
+### JavaScript-based Form Validation
+
+Form Validation is important to ensure that a user has submitted appropriate data. JavaScript offers an additional level of validation along with HTML native form attributes on the client side. Developers generally prefer validating form data through JavaScript because its data processing is faster when compared to server-side validation.
+
+Let's look at a simple example:
+
+```html
+<html>
+<body>
+<script>  
+function validateFormWithJS(){  
+var name=document.thisform.name.value;  
+var rollnumber=document.thisform.rollnumber.value;  
+  
+if (name==""){  
+  alert("Please fill out the name.");  
+  return false;  
+}else if(rollnumber.length<3){  
+  alert("Roll Number should be at least 3 digits long.");  
+  return false;  
+  }  
+}  
+</script>  
+<body>  
+<form name="thisform" method="post" action="/send-data-here" onsubmit="return validateFormWithJS()" >  
+<label for="name">Name:</label>
+<input type="text" name="name"><br/> 
+<label for="rollnumber">Roll Number:</label>
+<input type="text" name="rollnumber"><br/>
+
+<button type="submit">Submit</button>
+</form>  
+</body>
+</html>
+```
```

Comment:
Think we need to format this. Didn't prettier run?

Reply:
There is also an extra `<body>` in there. Shall we move the script to the bottom of the body (so form above the script), add some spacing between the form and script.

Reply:
Yeah, that extra body tag is preventing Prettier to work. Fixing that. 🙈